go/private/rules/rule.bzl

```diff
@@ -18,11 +18,14 @@ load(
 )
 
 def go_rule(implementation, attrs={}, toolchains=[], bootstrap=False, **kwargs):
+  attrs["_go_context_data"] = attr.label(default = Label("@io_bazel_rules_go//:go_context_data"))
+  aspects = []
+  if all([k in attrs for k in ["pure", "static", "msan", "race"]]):
```

Comment:
I don't find this condition intuitive, but I can't think of a better way to express it. Maybe just a comment explaining would help.